apps/prairielearn/src/lib/manualGrading.js

```diff
@@ -1,73 +1,134 @@
+  rubric_items_changed: z.boolean(),
+});
+
+/** @typedef {Partial<import('./db-types').RubricItem> & {order: number}} RubricItemInput */
+
 /** Builds the URL of an instance question tagged to be manually graded for a particular
  * assessment question. Only returns instance questions assigned to a particular grader.
  *
  * @param {string} urlPrefix - URL prefix for the current course instance.
- * @param {number} assessment_id - The assessment linked to the assessment question. Used to ensure the assessment is authorized, since middlewares don't authenticate assessment questions.
- * @param {number} assessment_question_id - The assessment question being graded.
- * @param {number} user_id - The user_id of the current grader. Typically the current effective user.
- * @param {number} prior_instance_question_id - The instance question previously graded. Used to ensure a consistent order if a grader starts grading from the middle of a list or skips an instance.
+ * @param {string} assessment_id - The assessment linked to the assessment question. Used to ensure the assessment is authorized, since middlewares don't authenticate assessment questions.
+ * @param {string} assessment_question_id - The assessment question being graded.
+ * @param {string} user_id - The user_id of the current grader. Typically the current effective user.
+ * @param {string} prior_instance_question_id - The instance question previously graded. Used to ensure a consistent order if a grader starts grading from the middle of a list or skips an instance.
  * @returns {Promise<string>}
  */
-async function nextUngradedInstanceQuestionUrl(
+export async function nextUngradedInstanceQuestionUrl(
   urlPrefix,
   assessment_id,
   assessment_question_id,
   user_id,
   prior_instance_question_id,
 ) {
-  const params = {
-    assessment_id,
-    assessment_question_id,
-    user_id,
-    prior_instance_question_id,
-  };
-  const result = await sqldb.queryZeroOrOneRowAsync(
+  const instance_question_id = await sqldb.queryOptionalRow(
     sql.select_next_ungraded_instance_question,
-    params,
+    { assessment_id, assessment_question_id, user_id, prior_instance_question_id },
+    IdSchema,
   );
 
-  if (result.rowCount > 0) {
-    const instance_question_id = result.rows[0].id;
-    return `${urlPrefix}/assessment/${assessment_id}/manual_grading/instance_question/${instance_question_id}`;
-  }
-  // If we have no more submissions, then redirect back to manual grading page
-  return `${urlPrefix}/assessment/${assessment_id}/manual_grading/assessment_question/${assessment_question_id}`;
+  return instance_question_id != null
+    ? `${urlPrefix}/assessment/${assessment_id}/manual_grading/instance_question/${instance_question_id}`
+    : // If we have no more submissions, then redirect back to manual grading page
+      `${urlPrefix}/assessment/${assessment_id}/manual_grading/assessment_question/${assessment_question_id}`;
```

Comment:
Nit, I find the non-ternary version of this to be easier to read given the inline comment here.